Remove unused parts of substitution list

valentin-krasontovitsch opened the pull request:
**Issue**
Closes #4168 and #4143


**Approach**
We try to move stuff from C to python, removing what is unneeded in the process.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

andreas-el commented:
I think this is ready to go.
What you should consider is providing a more descriptive PR header since this is ending up in the release notes.
Optionally squashing the commits, or at the very least altering the text for '[We always append to the substitution list.](https://github.com/equinor/ert/pull/4175/commits/894e43766b60b299a900dc8f0b4d6d16089feb38)'
Other than that, -- well done! :bacon: :rocket:

valentin-krasontovitsch commented:
updated the subject line of the append commit, changed the PR title to reflect the changes done more accurately and to be more descriptive - hope that's better! now we wait for the tests...

andreas-el commented:
Jenkins restest this please!
